ConsequencesTest/ConsoleWriterTest.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Diagnostics;
+using System.Security.Cryptography;
+using USACE.HEC.Hazards;
+using USACE.HEC.Results;
+using Xunit.Abstractions;
+namespace ConsequencesTest;
+public class ConsoleWrite
+{
+  static ResultItem r1 = new ResultItem { ResultName = "Depth", Result = 1.03f };
+  static ResultItem r2 = new ResultItem { ResultName = "Velocity", Result = 2.02f };
+  static ResultItem r3 = new ResultItem { ResultName = "ArrivalTime2ft", Result = new DateTime() };
+
+  static ResultItem[] resultItems = { r1, r2, r3 };
+  Result res = new Result(resultItems);
+  [Fact]
+  public void TestHeaders()
+  {
+    var stringWriter = new StringWriter();
```

Comment:
great addition in a unit test to verify behavior with an assert.

Another way you could do this is to write a "stringbuilderresultswriter" where the result is internally stored in a string builder. I like your approach because it can be utilized in many ways.